packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,611 @@
+import * as cdk from '@aws-cdk/core';
+import * as iam from '@aws-cdk/aws-iam';
+import * as kms from '@aws-cdk/aws-kms';
+import * as logs from '@aws-cdk/aws-logs';
+import { Metric, MetricOptions, Statistic } from '@aws-cdk/aws-cloudwatch';
+import { EbsDeviceVolumeType } from '@aws-cdk/aws-ec2';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import { LogGroupResourcePolicy } from './log-group-resource-policy';
+import { CfnDomain } from './elasticsearch.generated';
+
+export interface ClusterConfig {
+  readonly masterNodes: number;
+  readonly masterNodeInstanceType: string;
+  readonly dataNodes: number;
+  readonly dataNodeInstanceType: string;
+  /**
+   * The number of AZs that you want the domain to use. When you enable zone
+   * awareness, Amazon ES allocates the nodes and replica index shards that
+   * belong to a cluster across the specified number of Availability Zones (AZs)
+   * in the same region to prevent data loss and minimize downtime in the event
+   * of node or data center failure. Don't enable zone awareness if your cluster
+   * has no replica index shards or is a single-node cluster.
+   *
+   * @default - Zone awareness is not enabled.
+   */
+  readonly availabilityZoneCount?: number;
+}
+
+export interface EbsOptions {
+  readonly iops?: number;
+  readonly volumeSize: number;
+  readonly volumeType: EbsDeviceVolumeType;
+}
+
+export interface LoggingOptions {
+  /**
+   * Specify if slow search logging should be set up.
+   *
+   * @default - false
+   */
+  readonly slowSearchLogEnabed?: boolean;
+
+  /**
+   * Log slow searches to this log group.
+   *
+   * @default - a new log group is created if slow search logging is enabled
+   */
+  readonly slowSearchLogGroup?: logs.LogGroup;
+
+  /**
+   * Specify if slow index logging should be set up.
+   *
+   * @default - false
+   */
+  readonly slowIndexLogEnabed?: boolean;
+
+  /**
+   * Log slow indecies to this log group.
+   *
+   * @default - a new log group is created if slow index logging is enabled
+   */
+  readonly slowIndexLogGroup?: logs.LogGroup;
+
+  /**
+   * Specify if Elasticsearch application logging should be set up.
+   *
+   * @default - false
+   */
+  readonly appLogEnabed?: boolean;
+
+  /**
+   * Log Elasticsearch application logs to this log group.
+   *
+   * @default - a new log group is created if app logging is enabled
+   */
+  readonly appLogGroup?: logs.LogGroup;
```

Comment:
Can we change to `logs.ILogGroup` and support imported log groups as well?

Reply:
Changed.